apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx

```diff
@@ -785,6 +804,8 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
                  */
                 const grant: CheckboxChildWithIndex = {
                     label: modifyGrantTypeLabels(name, displayName),
+                    readOnly: (isM2MApplication && name === ApplicationManagementConstants.CLIENT_CREDENTIALS_GRANT) 
+                        ? true : false,
```

Comment:
can be simplified
```
isM2MApplication && name === ApplicationManagementConstants.CLIENT_CREDENTIALS_GRANT
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4605